sklearn/tests/test_metaestimators.py

```diff
@@ -119,45 +144,47 @@ def score(self, X, y, *args, **kwargs):
                if not k.startswith('_') and not k.startswith('fit')]
     methods.sort()
 
-    for delegator_data in DELEGATING_METAESTIMATORS:
-        delegate = SubEstimator()
-        delegator = delegator_data.construct(delegate)
-        for method in methods:
-            if method in delegator_data.skip_methods:
-                continue
-            assert hasattr(delegate, method)
-            assert hasattr(delegator, method), (
```

Comment:
For reviewers: these assert will fail for nested meta-estimator using delegation because methods will not exist before fit. I moved these assertions after calling `fit`.

Reply:
The remainder of the test is the same as before (only using parameterization to better debug failure).